__tests__/app.test.js

```diff
@@ -404,21 +402,50 @@ describe("DELETE /api/comments/:comment_id", () => {
   });
 });
 
-describe('GET /api/users', () => {
-  test('returns an array of topics objects with correct properties', () => {
-    return request(app)
-    .get("/api/users")
-    .expect(200)
-    .then((response) => {
-      const users = response.body.users;
-      expect(users.length).toBe(4);
-      users.forEach((user) => {
-        expect(user.hasOwnProperty("username")).toBe(true);
-        expect(user.hasOwnProperty("name")).toBe(true);
-        expect(user.hasOwnProperty("avatar_url")).toBe(true);
-
+describe("GET /api/users", () => {
+  test("returns an array of topics objects with correct properties", () => {
+    return request(app)
+      .get("/api/users")
+      .expect(200)
+      .then((response) => {
+        const users = response.body.users;
+        expect(users.length).toBe(4);
+        users.forEach((user) => {
+          expect(user.hasOwnProperty("username")).toBe(true);
+          expect(user.hasOwnProperty("name")).toBe(true);
+          expect(user.hasOwnProperty("avatar_url")).toBe(true);
+        });
+      });
+  });
+});
+describe("GET /api/articles?:topic_query", () => {
+  test("returns an array of articles with the specified topic value", () => {
+    return request(app)
+      .get("/api/articles?topic=mitch")
+      .expect(200)
+      .then((response) => {
+        const articles = response.body.articles;
+        expect(articles.length).toBe(12);
+        articles.forEach((article) => {
+          expect(article.topic).toBe("mitch");
+        });
+      });
+  });
+  test("returns 400 when topic does not exist", () => {
+    return request(app)
+      .get("/api/articles?topic=not-a-topic")
+      .expect(400)
+      .then((response) => {
+        expect(response.body.msg).toBe("bad request");
       });
-    });
   });
 
-})
+  test("returns 404 when topic exists but does not have any associated articles", () => {
+    return request(app)
+      .get("/api/articles?topic=paper")
+      .expect(404)
+      .then((response) => {
+        expect(response.body.msg).toBe("not found");
+      });
+  });
+});
```

Comment:
This test is close but we should actually check that the empty array is returned - it is not "not found" it's just there are no articles there, yet. As such it's not a 400 or 404 status